js/signin.js

```diff
@@ -0,0 +1,104 @@
+const emailInput = document.querySelector("#email-input");
+const passwordInput = document.querySelector("#password-input");
+const emailErrorMessage = document.querySelector("#email-error-message");
+const passwordErrorMessage = document.querySelector("#password-error-message");
+const loginForm = document.querySelector("#login-form");
+
+const eyeOffButton = document.querySelector(".eye-off");
+const eyeOnButton = document.querySelector(".eye-on");
+
+const EMAIL_REGEX = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,6}$/;
+
+// 이메일 입력
+emailInput.addEventListener("focusout", checkEmailInput);
+emailInput.addEventListener("focus", resetInputStyle);
+
+function checkEmailInput() {
+  if (emailInput.value.trim() === "") {
+    showEmailErrorMessage("이메일을 입력해 주세요.");
+  } else if (!validateEmail(emailInput.value.trim())) {
+    showEmailErrorMessage("올바른 이메일 주소가 아닙니다.");
+  } else {
+    hideErrorMessage();
+  }
+}
+
+function validateEmail(email) {
+  return EMAIL_REGEX.test(email);
+}
```

Comment:
`validateEmail` 이라는 함수명을 봤을 때 리턴값이 boolean일지 혹은 에러를 throw할지 조금 애매하게 느껴질 수 있을 것 같은데요. 
보통 조건을 검증해서 boolean(true, false)값을 return해주는 함수의 경우에는 is- prefix를 사용하는 편이라서 `isValidEmail`과 같은 이름은 어떨까 싶어요!
```suggestion
function isValidEmail(email) {
  return EMAIL_REGEX.test(email);
}
```

Reply:
추가적으로 이메일을 검증하기 위해 `EMAIL_REGEX.test(email)`과 같이 `checkEmailInput` 함수에서 사용할 수도 있는데요~ 
`validateEmail`이라는 이름이 오히려 이메일을 어떤 방식으로 검증하는지에 대한 정보를 숨기고 있기 때문에 헷갈릴만한 포인트가 있지 않을까? 에 대한 고민도 해보시면 좋을 것 같아요!
`추상화`에 대한 관점입니다!